projects/js-packages/charts/src/components/pie-chart/stories/donut.stories.tsx

```diff
@@ -319,3 +326,109 @@ export const CustomLegendPositioning: Story = {
 		},
 	},
 };
+
+const WooPieLegend = ( {
+	legendItems,
+	legendData,
+	withComparison = false,
+}: {
+	legendItems: BaseLegendItem[];
+	legendData: { label: string; value: number; formattedValue: string; comparison?: number }[];
+	withComparison?: boolean;
+} ) => (
+	<div
+		style={ {
+			display: 'inline-grid',
+			gridTemplateColumns: '1fr auto auto',
+			gap: 'var(--wpds-spacing-05, 5px) var(--wpds-spacing-10, 10px)',
+		} }
+	>
+		{ legendData.map( ( item, index ) => {
+			const { color } = legendItems[ index ];
```

Comment:
Extract the color from the legend items in the global charts context, which have a resolved color based on the theme, group, overrides, etc.